knora/dsplib/models/group.py

```diff
@@ -249,16 +249,14 @@ def getAllGroups(con: Connection) -> Optional[list[Group]]:
             return None
 
     @staticmethod
-    def getAllGroupsForProject(con: Connection, proj_shortcode: str) -> list[Group]:
-        result = con.get(Group.ROUTE)
-        if 'groups' not in result:
-            raise BaseError("Request got no groups!")
-        all_groups = result["groups"]
-        project_groups = []
-        for group in all_groups:
-            if group["project"]["id"] == "http://rdfh.ch/projects/" + proj_shortcode:
-                project_groups.append(group)
-        return list(map(lambda a: Group.fromJsonObj(con, a), project_groups))
+    def getAllGroupsForProject(con: Connection, proj_shortcode: str) -> Optional[list[Group]]:
+        all_groups: Optional[list[Group]] = Group.getAllGroups(con)
+        if all_groups:
+            project_groups = []
+            for group in all_groups:
+                if group.project == "http://rdfh.ch/projects/" + proj_shortcode:
+                    project_groups.append(group)
+            return project_groups
```

Comment:
Cool, you fixed DEV-786 even before the bug was reported :-) The e-mail exchange we had yesterday was about a researcher who encountered problems with exactly this :-)

Reply:
haha, true, I didn't even remember :D